### PR TITLE
Feature/add analytics

### PR DESCRIPTION
**Pull request by flofischer**
Added analytics support and player v8 support to the plugin
dashboard issue: https://github.com/bitmovin/dashboard/issues/1313

### CODE REVIEW

File: bitmovin.php [hunk 1]
```diff
@@ -356,13 +401,42 @@ function generate_player($id)
...
+        }
+        $html .= "conf.analytics = conf.analytics || {};\n";
+        $html .= "conf.analytics.key = '" . $analyticsKey . "';\n";
+    }
+
+    if($isOldPlayer){
+        $html .= "conf.source = source;\n";
+        $html .= "var player = bitmovin.player(\"bitmovin-player\");\n";
+        if($includeAnalytics == "on"){
+            $html .= "bitmovin.analytics.augment(player);\n";
```
Comment (lukaskroepfl) [inline bitmovin.php]: I think the .augment usage is deprecated/discouraged

File: bitmovin.php [hunk 2]
```diff
@@ -356,13 +401,42 @@ function generate_player($id)
...
+    $html .= "source.dash = '" . $dash . "';\n";
+    $html .= "source.hls = '" . $hls . "';\n";
+    $html .= "source.progressive = '" . $prog . "';\n";
+    $html .= "source.poster = '" . $poster . "';\n";
+
+    $analyticsKey = json_decode(get_post_meta($id, '_config_analytics_key', true));
+    $includeAnalytics = json_decode(get_post_meta($id, "_analytics_enabled", true)) == "on";
+
+    if($includeAnalytics){
+        $analytics_link = "https://cdn.bitmovin.com/analytics/web/1.7/bitmovinanalytics.min.js";
```
Comment (lukaskroepfl) [inline bitmovin.php]: use the latest `1`

File: bitmovin.php [hunk 3]
```diff
@@ -204,25 +218,47 @@ function getPlayerTable($id)
...
 
     $customTable = "<table class='wp-list-table widefat fixed striped'>";
 
     $customTable .= "<tr><td>Custom configuration</td><td><pre>var conf = {<br><div class='intend1'><textarea id='config_custom' name='config_custom_conf' placeholder='Insert your custom configuration here, e.g.:\nplayback: { \n    autoplay: true\n}\n'>" . $customConf . "</textarea></div>};</pre></td></tr>";
-    $customTable .= "<tr><td colspan='2' class='hint'>An Overview about all available configuration options can be found in our <a href='https://developer.bitmovin.com/hc/en-us/articles/115001689833' target='_blank'>documentation</a>. HINT: The configuration properties set above the custom configuration will override conflicts in this configuration.</td></tr>";
+    $customTable .= "<tr><td colspan='2' class='hint'>An Overview about all available configuration options can be found in our <a href='https://developer.bitmovin.com/hc/en-us/articles/115001689833' target='_blank'>documentation</a>. <br/> HINT: The configuration properties set above the custom configuration will override conflicts in this configuration.</td></tr>";
```
Comment (lukaskroepfl) [inline bitmovin.php]: old docs link, new: https://bitmovin.com/docs/player/api-reference/web/web-sdk-config-api-reference-v7

File: bitmovin.php [hunk 4]
```diff
@@ -204,25 +218,47 @@ function getPlayerTable($id)
...
+    $analytics_customConf = json_decode(get_post_meta($id, "_config_analytics_custom_conf", true));
+
+    $analyticsTable = '<table class="wp-list-table widefat fixed striped">';
+
+    $analyticsTable .= getTableRowSelect("Analytics Key", "config_analytics_key", $analytics_key, array());
+    $analyticsTable .= getTableRowInput("Video ID", "config_analytics_videoid", $analytics_videoid, "Video id");
+    $analyticsTable .= "<tr><td>Custom configuration</td><td><pre>var analyticsConf = {<br><div class='intend1'><textarea id='config_analytics_custom' name='config_analytics_custom_conf' placeholder='Insert your custom configuration here, e.g.:\n title: \"A descriptive video title\"\n'>" . $analytics_customConf . "</textarea></div>};</pre></td></tr>";
+    $analyticsTable .= "<tr><td colspan='2' class='hint'>An Overview about all available configuration options can be found in our <a href='https://developer.bitmovin.com/hc/en-us/articles/115001689833' target='_blank'>documentation</a>. <br/>HINT: The configuration properties set above the custom configuration will override conflicts in this configuration.</td></tr>";
```
Comment (lukaskroepfl) [inline bitmovin.php]: pls remove that `An Overview about all available configuration...` sentence as it's for player

File: bitmovin.php (hunk 1 above)
Comment (SInCE) [reply bitmovin.php]: @Tigraine please comment (https://bitmovin.com/docs/analytics/quickstarts/integrating-bitmovin-analytics-with-bitmovin-player) states the usage of augment for that

Comment (Tigraine) [reply bitmovin.php]: will update docs..

File: bitmovin.php (hunk 2 above)
Comment (Tigraine) [reply bitmovin.php]: please use latest 2 :)

File: bitmovin.php (hunk 1 above)
Comment (Tigraine) [reply bitmovin.php]: We moved to a new syntax with v2 .. i'll add docs tomorrow and update the PR myself

Comment (lukaskroepfl) [reply bitmovin.php]: any update @Tigraine ?

File: bitmovin.php [hunk 5]
```diff
@@ -356,13 +399,42 @@ function generate_player($id)
...
+    $html .= "source.dash = '" . $dash . "';\n";
+    $html .= "source.hls = '" . $hls . "';\n";
+    $html .= "source.progressive = '" . $prog . "';\n";
+    $html .= "source.poster = '" . $poster . "';\n";
+
+    $analyticsKey = json_decode(get_post_meta($id, '_config_analytics_key', true));
+    $includeAnalytics = json_decode(get_post_meta($id, "_analytics_enabled", true)) == "on";
+
+    if($includeAnalytics){
+        $analytics_link = "https://cdn.bitmovin.com/analytics/web/2.0/bitmovinanalytics.min.js";
```
Comment (Tigraine) [inline bitmovin.php]: I'd honestly change this to be `/2/`:
`https://cdn.bitmovin.com/analytics/web/2/bitmovinanalytics.min.js`

File: bitmovin.php [hunk 6]
```diff
@@ -356,13 +399,42 @@ function generate_player($id)
...
+        }
+        $html .= "conf.analytics = conf.analytics || {};\n";
+        $html .= "conf.analytics.key = '" . $analyticsKey . "';\n";
+    }
+
+    if($isOldPlayer){
+        $html .= "conf.source = source;\n";
+        $html .= "var player = bitmovin.player(\"bitmovin-player\");\n";
+        if($includeAnalytics == "on"){
+            $html .= "bitmovin.analytics.augment(player);\n";
```
Comment (Tigraine) [inline bitmovin.php]: With v2 of the collector the correct usage is simply:

`new bitmovin.analytics.adapters.Bitmovin7Adapter(player);`

Comment (Tigraine) [inline bitmovin.php]:
```suggestion
            $html .= "bitmovin.analytics.augment(player);\n";
            $html .= "new bitmovin.analytics.adapters.Bitmovin7Adapter(player);\n";
```

File: bitmovin.php [hunk 7]
```diff
@@ -356,13 +399,42 @@ function generate_player($id)
...
+        }
+        $html .= "player.setup(conf).then(function(value) {\n";
+    }
+    else {
+        $html .= "var container = document.getElementById('bitmovin-player');\n";
```
Comment (Tigraine) [inline bitmovin.php]:
```suggestion
        $html .= "var container = document.getElementById('bitmovin-player');\n";
        $html .= "bitmovin.player.Player.addModule(bitmovin.analytics.PlayerModule);\n";
```

To get the latest analytics version that came from CDN and not use the one packaged in the player this call is required

**Review by Tigraine — CHANGES_REQUESTED** [review]
@lukaskroepfl added clarification on API etc

**Review by Tigraine — APPROVED** [review]
👍

File: js/bitmovin.js [hunk 8]
```diff
@@ -21,6 +21,18 @@ function collapseAdvancedPanels() {
       $j("#" + configSections[i]).addClass("closed");
     }
   }
+  if(!document.getElementById('analytics_enabled').checked){
+      $j("#bitmovin_player_configuration_analytics").addClass("closed");
+  }
+}
+
+function handleAnalyticsCheckboxChange(checkbox) {
```
Comment (SInCE) [inline js/bitmovin.js:27]: When Analytics is `disabled` also disable analytics input fields so they can't be edited, unless the checkbox is set to make it more clear that this configuration has no effect if disabled

**Review by SInCE — CHANGES_REQUESTED** [review]

File: js/bitmovin.js (hunk 8 above)
Comment (SInCE) [reply js/bitmovin.js:27]: Please move the 'Enable Analytics' button as well into the analytics configuration section. @flofischer

**Review by lukaskroepfl — APPROVED** [review]

**Review by SInCE — APPROVED** [review]

**Pull request merged**